### ISSUE TITLE
NuGet Restore Task Fails sometime using -DisableParallelProcessing helps

### ISSUE THREAD

**Issue by johnkattenhorn** [Microsoft/vso-agent-tasks#672]
It's 2AM and I know this sounds strange but I've been working on this all day. 

We have a .sln which has been building for months without the need for the separate Task for NuGet Restore; we just used to tick the box on the build task to perform the restore.

Then something happened, I think someone introduced a 4.5.2 project which pulled in some packages which would not restore. They came back with a message along the lines of you must use NuGet3 to install System.Linq.xxx.

So I added NuGet 3.x to the git repo and added an additional NuGet Restore Task, again everything was fine. Then after a few days I start to get restore failures,
 
```Unable to find version '3.3.3' of package 'Castle.Core'.
NuGet.CommandLineException: Exception of type 'NuGet.CommandLineException' was thrown.
at NuGet.CommandLine.RestoreCommand.<ExecuteCommandAsync>d__18.MoveNext()
Unexpected exit code 1 returned from tool nuget.exe```

So after a while I found that the projects which were 4.5.2 are now 4.5.1 but they are referencing some NuGet packages which seem to only work with 4.5.2 (can't remember the whole list but things like `System.Runtime`.

So I removed these packages and the build starts working again for a few hours until, stops working again, same failure as before, sometimes `Autofac` or `Autofac.DynamicProxy` but the reason of the message is the same.

The problem becomes serious, every single build then starts to fail as of today and the only way I can get a build to work sometimes is with `-NoCache -DisableParallelProcessing` but fail more often that not.

Can anyone suggest how I might be able to fault find the route cause of this problem ? Anybody seen anything like this before ?

We have a couple of other team VSO subscriptions which have similar configurations to this one in terms of steps but they are absolutely fine.

**Comment by chrisrpatterson** [comment]
Response from the NuGet team @yishaigalatzer

We are working on some similar issues. But the immediate question is the user is using 3.2.1-RC that resolves some of these already.

**Comment by johnkattenhorn** [comment]
Hey @chrisrpatterson , I'm using 3.2.0.10516, sounds like I should give the RC a go :-)

Thanks for the rapid response, I'll let you what I find.

**Comment by vtbassmatt** [comment]
@johnkattenhorn did you get to the bottom of this? We've got 3.2.1 on the build agents now, and of course you're free to check in whatever nuget.exe works for your scenario.

**Comment by yishaigalatzer** [comment]
@vtbassmatt current version is 3.3

**Comment by vtbassmatt** [comment]
@yishaigalatzer we're still figuring out our update strategy for NuGet on the hosted agents. Some have wanted to always be bleeding-edge while others want to pin a known-good version. At the time of writing, we have 3.2.1 on the hosted agents. We should look at moving to 3.3, though. Thanks!

**Comment by yishaigalatzer** [comment]
I assume you also support "bring your own NuGet"?

**Comment by vtbassmatt** [comment]
Yes :)

**Comment by vtbassmatt** [comment]
Since we have 3.2.1 on the build agents and I haven't heard any more problems, closing this issue.

**Comment by yishaigalatzer** [comment]
@vtbassmatt what is the plan to update to 3.3? 3.2.1 is definitely out of date and I don't see why it shouldn't be the default

**Comment by vtbassmatt** [comment]
@yishaigalatzer there isn't a plan. I'm not aware of any complaints that 3.3 would fix. We're close enough to 3.4 that we'll at least wait for that release.

**Comment by yishaigalatzer** [comment]
I disagree with that, 3.4 is not close it is months away. Look at the bug list for 3.3, and you will see it is not small. Last if we don't get feedback for 3.3 we won't fix it in 3.4

**Comment by cherici-simone** [comment]
In these days I have the same problem when trying to restore packages from https://XXX.pkgs.visualstudio.com/DefaultCollection/_packaging/XXX/nuget/v3/index.json

the error is:
2016-03-08T11:32:31.4143065Z ##[error]Unable to find version '2016.2.25.14-RC-201622514' of package 'MRCS.SmartGateway.Activities'.
2016-03-08T11:32:31.4153071Z ##[error]Unable to find version '1.0.51113' of package 'MRCS.SmartSuite.WFMCore'.
2016-03-08T11:32:31.4683069Z ##[error]Unexpected exit code 1 returned from tool NuGet.exe
